src/functions/casing.ts

```diff
@@ -36,6 +36,16 @@ export const casing: IFunction<ICasingOptions> = (targetVal, opts) => {
     return;
   }
 
+  if (
+    targetVal.length === 1 &&
+    opts.separator !== void 0 &&
+    opts.separator.allowLeading === true &&
+    // eslint-disable-next-line @typescript-eslint/prefer-string-starts-ends-with
+    targetVal[0] === opts.separator.char
```

Comment:
@P0lip There might be something I don't understand (and in that case I apologize in advance for being _that_ thick), but we know targetVal is one char long, we also know the separator is also one char long (enforced through schema rule). Can't we just compare them directly (ie. `targetVal === opts.separator.char`)?

Reply:
well, yeah, foolish me :man_facepalming: